elisa/src/org/labkey/elisa/HighThroughputImportHelper.java

```diff
@@ -0,0 +1,304 @@
+        {
+            if (_stdConcentrations.containsKey(analyteNum))
+                return _stdConcentrations.get(analyteNum);
+            else
+                return Collections.emptyMap();
+        }
+
+        public Map<Integer, double[][]> getDataMap()
+        {
+            return _dataMap;
+        }
+
+        public String getPlateName()
+        {
+            return _plateName;
+        }
+
+        public void setStdConcentration(Position position, Integer spot, Double concentration)
+        {
+            for (WellGroupTemplate wellGroup : _plateTemplate.getWellGroups(position))
+            {
+                if (wellGroup.getType() == WellGroup.Type.REPLICATE)
+                {
+                    Map<String, Double> concentrations = _stdConcentrations.computeIfAbsent(spot, s -> new HashMap<>());
+                    concentrations.put(wellGroup.getPositionDescription(), concentration);
+
+                    return;
+                }
+            }
+        }
+
+        public void setRawSignal(Position position, Integer spot, Integer signal)
+        {
+            if (position != null && spot != null && signal != null)
+            {
+                double[][] data = _dataMap.computeIfAbsent(spot, s -> new double[_plateTemplate.getRows()][_plateTemplate.getColumns()]);
+                data[position.getRow()][position.getColumn()-1] = signal.doubleValue();
+            }
+        }
+
+        public void setExtraProperties(Map<String, Object> row, PositionImpl position, Integer spot, List<? extends DomainProperty> resultsDomain)
+        {
+            Map<String, Object> extraProperties = new HashMap<>();
+            // need to adjust the column value to be 0 based to match the template locations
+            position.setColumn(position.getColumn()-1);
+
+            for (WellGroupTemplate wellGroup : _plateTemplate.getWellGroups(position))
+            {
+                if (wellGroup.getType() == WellGroup.Type.CONTROL)
+                {
+                    // get the control ID for control well groups
+                    if (row.containsKey(CONTROL_ID_COLUMN))
+                        extraProperties.put(ElisaAssayProvider.CONTROL_ID_PROPERTY, row.get(CONTROL_ID_COLUMN));
+
+                    break;
+                }
+            }
+
+            // pick up any properties that match those in the results domain (outside of the built-in ones)
+            for (DomainProperty prop : resultsDomain)
+            {
+                if (!ElisaAssayProvider.REQUIRED_RESULT_PROPERTIES.contains(prop.getName()))
+                {
+                    Object value = getValue(row, prop);
+                    if (value != null)
+                        extraProperties.put(prop.getName(), value);
+                }
+            }
+            _extraWellData.put(getWellAnalyteKey(position, spot), extraProperties);
+        }
+
+        private Object getValue(Map<String, Object> row, DomainProperty property)
```

Comment:
Do we have any sort of centralized place that does this "get value for a property name or import aliases"? It seems like something we'd possibly be doing all over the place when we have a row Map and a DomainProperty.